x-pack/plugins/security_solution/public/timelines/components/side_panel/event_details/index.tsx

```diff
@@ -106,6 +106,12 @@ const EventDetailsPanelComponent: React.FC<EventDetailsPanelProps> = ({
     return endpointAlertCheck({ data: detailsData || [] });
   }, [detailsData]);
 
+  const ruleName = useMemo(() => {
+    const findRuleName = find({ category: 'signal', field: 'signal.rule.name' }, detailsData)
+      ?.values;
+    return findRuleName ? findRuleName[0] : '';
```

Comment:
Could `findRuleName` be an empty array? Would it be better to check `findRuleName?.length`. Also (nit), it might be helpful to rename this to `currentRuleName` or something similar

Reply:
ok, I'll apply that check to other fields as well.